src/Symfony/Component/Yaml/Dumper.php

```diff
@@ -69,9 +69,9 @@ public function dump($input, int $inline = 0, int $indent = 0, int $flags = 0):
                 }
 
                 if (Yaml::DUMP_MULTI_LINE_LITERAL_BLOCK & $flags && \is_string($value) && false !== strpos($value, "\n") && false === strpos($value, "\r")) {
-                    // If the first line starts with a space character, the spec requires a blockIndicationIndicator
+                    // If the first non-empty line starts with a space character, the spec requires a blockIndicationIndicator
```

Comment:
Is that true? Or what happens when we change the fixture from your test to something like this?

```php
$data = [
    'data' => [
        'multi_line' => "\n    \n    the first line has leading newline and spaces\nThe second line does not.",
    ],
];
```

Reply:
The parser treats any line containing only spaces as an empty line, even if the number of space exceeds the indentation.

So it isn't enough to trim off the first newline before looking for a leading space, the dumper needs to find the first line that is neither empty nor contains only spaces.

I've added an additional test to check that the indicator is not added if the value begins with a line containing _only_ spaces.

The indentation indicator check is now more than a single-line operation, so I've moved it into a private function.